api/v1/storagecluster_types.go

```diff
@@ -105,6 +105,10 @@ type StorageClusterSpec struct {
 	DefaultStorageProfile string `json:"defaultStorageProfile,omitempty"`
 
 	StorageProfiles []StorageProfile `json:"storageProfiles,omitempty"`
+
+	// EnableLoadBalancerAPIServerService Indicates if the OCS Provider API Server Service should be
+	// of type NodePort or LoadBalancer.
+	EnableLoadBalancerAPIServerService bool `json:"enableLoadBalancerAPIServerService,omitempty"`
```

Comment:
Making the flag a boolean value can cause issues with upgrades since the existing clusters have the service configured as a load balancer by default, and the flag's value is set to false. This will change the service type from load balancer to nodeport on the existing clusters, which is not desired.

To avoid this, I propose using a string instead of a boolean. By default, if the string is empty, we will create a load balancer service. This approach ensures that we can easily introduce new service types in the future without requiring any modifications to the API.

Furthermore, can you please move this flag right after the AllowRemoteStorageConsumers flag. Additionally, in the comment, mention that this flag will only be considered if AllowRemoteStorageConsumers is set to true.